File: 30 Days of Code/11 - Day 10 - Binary Numbers.py
# ========================
#       Information
# ========================

# Direct Link: https://www.hackerrank.com/challenges/30-binary-numbers/problem
# Difficulty: Easy
# Max Score: 30
# Language: Python

# ========================
#         Solution
# ========================

N = int(input())
DATA = bin(N)

MAXIMUM = 0
CURRENT = 0

for num in DATA:
    if num == '1':
        CURRENT = CURRENT + 1
    else:
        MAXIMUM = max(MAXIMUM, CURRENT)
        CURRENT = 0

print(max(MAXIMUM, CURRENT))

File: 30 Days of Code/C#/11 - Day 10 - Binary Numbers.cs
// ========================
//       Information
// ========================

// Direct Link: https://www.hackerrank.com/challenges/30-binary-numbers/problem
// Difficulty: Easy
// Max Score: 30
// Language: C#

// ========================
//         Solution
// ========================

using System;

class Solution {
	static void Main(String[] args) {
		var n = int.Parse(Console.ReadLine());

		var sum = 0;
		var max = 0;

		while (n > 0) {
			if (n % 2 == 1) {
				sum++;

				if (sum > max) {
					max = sum;
				}
			}
			else {
				sum = 0;
			}
			n = n / 2;
		}
		Console.WriteLine(max);
	}
}
